pthread/pthread.c

```diff
@@ -183,6 +191,22 @@ static void pthread_full_key_cleanup(_pthread_key_cleanup_t *head)
 }
 
 
+static void _pthread_do_cleanup(pthread_ctx *ctx)
+{
+	pthread_cleanup_t *head = ctx->cleanup_list;
+	while (ctx->cleanup_list != NULL) {
```

Comment:
Doesn't really matter, just seems more right to me, just a suggestion
```suggestion
	while (head != NULL) {
```